runtime/utils/Python.hpp

```diff
@@ -0,0 +1,3 @@
+#include <mutex>
+
+std::mutex &getPythonMutex();
```

Comment:
Is this header useful? (I saw you declared the function manually in the openqasm runner anyways)

Reply:
Ah, then I should delete it from the openqasm runner. That is its only use.

Reply:
yeah, this can be removed.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/615/commits/4835ce7138f7610a3b39d69ec8a20346bf903783